services/src/kilda-pce/src/main/java/org/openkilda/pce/impl/BestCostAndShortestPathFinder.java

```
@@ -155,45 +157,40 @@ public BestCostAndShortestPathFinder(int allowedDepth, int defaultLinkCost) {
-                to = hint.get(hint.size() - 1).getDestSwitch();
-            }
-
-            // If the start/end equals from/to, then confirm the path and see if we can use it.
-            if (start.equals(from) && end.equals(to)) {
-                log.trace("getPath w/ Hint: hint matches start/ed {}->{}", from, to);
-                // need to validate that the path exists .. and if so .. set bestCost and bestPath
-                SearchNode best = confirmIsls(hint);
-                if (best != null) {
-                    log.debug("getPath w/ Hint: the hint path EXISTS for {}->{}", from, to);
-                    hintedReverseCost = best.parentCost;
-                    hintedReversePath = best;
+        if (forwardPath != null && !forwardPath.isEmpty()) {
+            List<Isl> reversePath = Lists.reverse(forwardPath);
+            reversePath = swapSrcDst(reversePath);
+
+            if (isPathEndpointsCorrect(src, dst, reversePath)) {
+                SearchNode symReversePath = confirmIsls(reversePath);
+                if (symReversePath != null) {
+                    log.debug("Reverse path is available from {} to {}", src, dst);
+                    return symReversePath.parentPath;
                 } else {
-                    log.info("getPath w/ Hint: the hint path DOES NOT EXIST for {}->{}, will find new path",
-                            from, to);
+                    log.warn("Failed to find reverse path from {} to {}, will find new alternative path", src, dst);
                 }
             }
         }
 
-        List<Isl> altReversePath = getPath(start, end, hintedReverseCost);
-        return altReversePath.isEmpty() && hintedReversePath != null ? hintedReversePath.parentPath : altReversePath;
+        List<Isl> alternativePath = getPath(src, dst);
+        if (alternativePath.isEmpty()) {
+            throw new UnroutableFlowException("Failed to find reverse path. Wasn't able to find an alternative path");
```

It will be better to throw this exception from `getPath(...)`.